tests/engine/adhoc_test.go

```diff
@@ -221,3 +227,83 @@ func importedAndExportedFunc(t *testing.T, newEngine func() wasm.Engine) {
 	require.NoError(t, err)
 	require.Equal(t, uint64(42), results[0])
 }
+
+//  hostFuncWithFloatParam fails if a float parameter corrupts a host function value
+func hostFuncWithFloatParam(t *testing.T, newEngine func() wasm.Engine) {
+	ctx := context.Background()
+	watTemplate, err := template.New("").Parse(`(module
+		;; '{{ . }}' will be replaced by the type name, f32 or f64.
+
+		;; 'identity_float' just returns the given float value as is.
+		(import "test" "identity_float" (func $test.identity_float (param {{ . }}) (result {{ . }})))
+
+		;; 'call->test.identity_float' proxies 'test.identity_float' via call in order to test floats aren't corrupted 
+		;; when the float values are passed through OpCodeCall
+		(func $call->test.identity_float (param {{ . }}) (result {{ . }})
+			local.get 0
+			call $test.identity_float
+		)
+		(export "call->test.identity_float" (func $call->test.identity_float))
+		)`)
+	require.NoError(t, err)
+
+	tests := []struct {
+		testName          string
+		typeName          string        // "f32" or "f64"
```

Comment:
maybe export `valueTypeName` then you can use `ValueType` here instead of a string (you can use `valueTypeName(vt)` at the call site I mean

Reply:
improved. Thanks.